### PR TITLE
Fixed optimized (wrapped) percentiles

### PR DESCRIPTION
**Pull request by oxtoacart**
This fix allows queries to use PERCENTILE expressions that wrap existing PERCENTILEs.

### CODE REVIEW

File: expr/percentile_optimized.go [hunk 1]
```diff
@@ -2,74 +2,40 @@ package expr
...
 import (
 	"fmt"
-	"time"
 
-	"github.com/getlantern/goexpr"
 	"github.com/getlantern/msgpack"
 )
 
 // PERCENTILEOPT returns an optimized PERCENTILE that wraps an existing
-// PERCENTILE and reuses its storage.
```
Comment (oxtoacart) [inline expr/percentile_optimized.go]: We no longer attempt to reuse the existing percentile's storage, we assume separate storage for this percentile.

File: expr/percentile_optimized.go [hunk 2]
```diff
@@ -2,74 +2,40 @@ package expr
...
 }
 
 type ptileOptimized struct {
+	ptile
 	Wrapped    Expr
-	wrapped    *ptile
 	Percentile Expr
 }
 
-func (e *ptileOptimized) Validate() error {
```
Comment (oxtoacart) [inline expr/percentile_optimized.go]: Most of this stuff is now inherited from `ptile`.

File: zenodb_test.go [hunk 3]
```diff
@@ -463,6 +465,27 @@ ORDER BY _time`
 	})
 }
 
+func testPercentileOptimizedQuery(wg *sync.WaitGroup, t *testing.T, db *DB, includeMemStore bool, epoch time.Time, resolution time.Duration) {
+	defer wg.Done()
+
+	sqlString := `
+SELECT PERCENTILE(pp_5p, 90) AS pp_opt
```
Comment (oxtoacart) [inline zenodb_test.go]: This shows an example of what's now possible.

Comment (hwh33) [reply zenodb_test.go]: Cool!  Thanks also, the context is helpful.

**Comment by oxtoacart** [comment]
Thanks @hwh33!

**Pull request merged**